guppylang/std/reflection.py

```diff
@@ -0,0 +1,10 @@
+"""Utilities for compile-time reflection."""
+
+# mypy: disable-error-code="empty-body, misc, override, valid-type, no-untyped-def"
+
+from guppylang.decorator import guppy
+from guppylang.std._internal.checker import CallableChecker
+
+
+@guppy.custom(checker=CallableChecker(), higher_order_value=False)
+def callable(x): ...
```

Comment:
We could also consider to just turn this into unsupported, this function is pretty useless...